fw/hpack.c

```diff
@@ -1435,7 +1477,8 @@ tfw_hpack_decode(TfwHPack *__restrict hp, unsigned char *__restrict src,
 			       __func__, hp->index);
 			WARN_ON_ONCE(!hp->index);
 			entry = tfw_hpack_find_index(&hp->dec_tbl, hp->index);
-			if (!entry || tfw_hpack_hdr_name_set(hp, req, entry)) {
+			if (!entry || !tfw_hpack_entry_tag_valid(entry->tag) ||
+				tfw_hpack_hdr_name_set(hp, req, entry)) {
```

Comment:
`tfw_hpack_entry_tag_valid()` is always and only called after `tfw_hpack_find_index()`, so I propose to call it from `tfw_hpack_find_index()` and make the function not to return invalid `entry`